modules/demo/api-server/routes/graphology/index.js

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) 2022, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const Fs = require('fs');
+const {Utf8String, Int32, Uint32, Float32, DataFrame, StringSeries, Series, Float64} =
+  require('@rapidsai/cudf');
+const {RecordBatchStreamWriter, Field, Vector, List, Table} = require('apache-arrow');
+const Path                                                  = require('path');
+const {promisify}                                           = require('util');
+const Stat                                                  = promisify(Fs.stat);
+const fastifyCors                                           = require('@fastify/cors');
+const fastify                                               = require('fastify');
+
+const arrowPlugin                                      = require('fastify-arrow');
+const gpu_cache                                        = require('../../util/gpu_cache.js');
+const {collapseTextChangeRangesAcrossMultipleVersions} = require('typescript');
```

Comment:
Need advice about ordering and arranging `requires` here. You said once to go from most general to lease general? What about alphabetical?

Reply:
The import order shouldn't really matter in JS, In C++ that the include order should go from most specific -> most general, because `#include` directives in the source file can mask missing includes. For example:

```c++
// main_bad.cpp
#include <map>
#include "main.hpp"
```
```c++
// main.hpp
// If someone includes `main.hpp`, the following line will error:
using some_type = std::map<std::string, int32_t>;

// It doesn't in `main_bad.cpp`, because it has an `#include <map>`
// before the `#include main.hpp`. This is bad, because the
// missing include in this file should be surfaced as an error!
```

```c++
// main_good.cpp
#include "main.hpp" // Now the compiler will tell us if `main.hpp` is missing an include
#include <map>
```

Reply:
Whoa. I'm infuriated that I never realized this. I thought of the includes across all the files being pushed to the top during precompile a sort of feature. :bear: